cachito/workers/pkg_managers/gomod.py

```diff
@@ -187,7 +189,47 @@ def resolve_gomod(app_source_path, request, dep_replacements=None):
             )
             shutil.copytree(tmp_download_cache_dir, str(bundle_dir.gomod_download_dir))
 
-        return module, deps
+        # package level dependencies
+        go_list_deps_output = run_gomod_cmd(
+            ("go", "list", "-deps", "-f", "{{if not .Standard}}{{.ImportPath}} {{.Module}}{{end}}"),
+            run_params,
+        )
+
+        pkg_level_deps = []
+        for line in go_list_deps_output.splitlines():
+            # line example: pkg.io/foo/bar pkg.io/foo/var v1.0.0 => pkg.io/foo/bar v1.0.1
+            parts = [part for part in line.split(" ") if part not in ("")]
+            replaces = None
+
+            name = parts[0]
+            parent_module = parts[1]
+            if len(parts) > 2:
+                version = parts[2]
+            else:
+                # use requested module version
+                version = module_version
+
+            if len(parts) > 3:
+                # parts[3] is just an arrow "=>" pointing to the replacements
+                replaces = {"type": "gomod", "name": parent_module, "version": version}
```

Comment:
@athos-ribeiro I'm wondering if we should just not set replaces for Go packages since the package isn't being replaced, only the module is. If you want to know if the Go package was from a replaced module, you should be able to check that from: go package -> go module -> replaces. I may be misunderstanding this though.

Reply:
By the way, if you want to keep this away, I'm also okay with that.